rollbar/lib/telemetry.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import threading
+import time
+
+import rollbar
+
+
+class Queue:
+    def __init__(self, size):
+        self.size = size
+        self.items = []
```

Comment:
in the constructor ?

Reply:
Not in the constructor but in the `__init__()` method. The current implementation leads to race conditions.

Reply:
This is safe as is, isn't it?

Reply:
well, __init__() is a constructor to me. Queue is initialized only once. no race conditions.

Reply:
This is not safe in CPython prior to version 3.3 as importing modules is not thread-safe (see [PEP 302](https://www.python.org/dev/peps/pep-0302/) and [the issue](https://bugs.python.org/issue9260)). Also, `__init__()` is not a constructor but an initializer (`__new__()` is a constructor to Python) and it's not thread-safe too:
https://github.com/python/cpython/blob/9816777861a3f19ea9faa5b98c9b7024f13c6b73/Objects/object.c#L180-L189

However, as mentioned earlier, this is not _a big deal_ as the race condition can occur only in the very rare cases of multi-loading the module by concurrent threads. It is rather against Python principles, so it can be handled by the user.

PS. How can `__init__()` be a constructor to Python if the method signature requires `self` as the first argument? ;)

Reply:
python got it so wrong, that's why everybody calls __init__() constructor. Have you ever seen somebody do this awkward init ? 
```
class A(object):
    def __new__(cls, *arg):
         cls.n = arg[0]
         return super(A, cls).__new__(cls)


    def __init__(self,_):
        self.n += 1

a = A(3)
print a.n

```